### PR TITLE
Particles tutorial

### PR DESCRIPTION
**Pull request by caguero**
A tutorial describing how to use particles.

### CODE REVIEW

File: tutorials/20_particles_tutorial.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+emitter->SetRate(10);
+emitter->SetParticleSize({1, 1, 1});
+emitter->SetLifetime(2);
+emitter->SetVelocityRange(10, 20);
+emitter->SetMaterial(particleMaterial);
+emitter->SetColorRangeImage(RESOURCE_PATH + "/smokecolors.png");
+emitter->SetScaleRate(10);
+emitter->SetEmitting(true);
+root->AddChild(emitter);
+```
```
Comment (iche033) [inline tutorials/20_particles_tutorial.md:94]: you can add tags to wrap a code snippet in the `particles_demo` example and refer to it in the tutorial. Here's [an example](https://github.com/ignitionrobotics/ign-rendering/blob/master/examples/render_pass/Main.cc#L155) of a tag, and to to refer to it:

```
\snippet examples/particles_demo/Main.cc your tag
```

File: tutorials/20_particles_tutorial.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+mkdir build
+cd build
+cmake ..
+make
+```
+Execute the example:
+
+```{.sh}
+./particles_demo
+[Msg] Loading plugin [ignition-rendering4-ogre]
```
Comment (iche033) [inline tutorials/20_particles_tutorial.md:20]: this should be `ogre2`? i.e. `ignition-rendering4-ogre2`

File: tutorials/20_particles_tutorial.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+Engine 'optix' is not supported
```
Comment (iche033) [inline tutorials/20_particles_tutorial.md:21]: how about we remove `optix` from this demo so this msg does not get printed out? so mainly [this line](https://github.com/ignitionrobotics/ign-rendering/blob/master/examples/particles_demo/Main.cc#L151)

File: tutorials/20_particles_tutorial.md (hunk 1 above)
Comment (caguero) [reply tutorials/20_particles_tutorial.md]: Thanks, done in 3a233f6.

File: tutorials/20_particles_tutorial.md (hunk 2 above)
Comment (caguero) [reply tutorials/20_particles_tutorial.md]: Yup, fixed in 3a233f6.

File: tutorials/20_particles_tutorial.md (hunk 3 above)
Comment (caguero) [reply tutorials/20_particles_tutorial.md:21]: Agreed! Done in 3a233f6.

**Comment by caguero** [comment]
> Don't forget to add the tutorial to https://github.com/ignitionrobotics/ign-rendering/blob/master/tutorials.md.in

Added in 3a233f6.

**Comment by iche033** [comment]
minor tweak: removed optix warning msg in 012dae8

**Pull request merged**